### PR TITLE
ECER-1031: character references

### PR DESCRIPTION
**Pull request by SoLetsDev**
## Title
ECER-1031: BE - FE integration for saving character references.

## Checklist
- [x] I have tested these changes locally.
- [ ] Changes to backend endpoints are covered by passing integration tests.
- [ ] I have added or updated the necessary documentation.

## Screenshots (if applicable)
image

### CODE REVIEW

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/wizard.ts [hunk 1]
```diff
@@ -84,11 +84,15 @@ export const useWizardStore = defineStore("wizard", {
...
-        [wizard.steps.characterReference.form.inputs.phoneNumber.id]: "",
+        // Character reference characterReference data from draft application
+        [wizard.steps.characterReferences.form.inputs.firstName.id]:
+          draftApplication.characterReferences && draftApplication.characterReferences[0] ? draftApplication.characterReferences[0].firstName : null,
+        [wizard.steps.characterReferences.form.inputs.lastName.id]:
+          draftApplication.characterReferences && draftApplication.characterReferences[0] ? draftApplication.characterReferences[0].lastName : null,
+        [wizard.steps.characterReferences.form.inputs.email.id]:
+          draftApplication.characterReferences && draftApplication.characterReferences[0] ? draftApplication.characterReferences[0].emailAddress : null,
+        [wizard.steps.characterReferences.form.inputs.phoneNumber.id]:
+          draftApplication.characterReferences && draftApplication.characterReferences[0] ? draftApplication.characterReferences[0].phoneNumber : null,
```
Comment (SoLetsDev) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/wizard.ts:95]: dual check for characterReferences then accessing the 1st index needed when we do not have a draft application. 

accessing charRef[0] index would result in an exception.

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Application.vue [hunk 2]
```diff
@@ -93,6 +93,9 @@ export default defineComponent({
           case "Education":
           case "WorkReferences":
           case "CharacterReferences":
```
Comment (pcatkins) [inline src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Application.vue:95]: Can we combine?

```
case "CharacterReferences":
case "Review":
            this.saveDraftAndAlertSuccess();
            this.incrementWizard();
```

Comment (SoLetsDev) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/pages/Application.vue:95]: Ahh nice catch. I didn't realize we were combining all the cases.

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/wizard.ts (hunk 1 above)
Comment (pcatkins) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/wizard.ts:95]: NITPICK: Same thing but a little less code:

```
[wizard.steps.characterReferences.form.inputs.firstName.id]:
  draftApplication.characterReferences?.[0]?.firstName ?? null,
```

Would empty string be better for wizard data?

```
[wizard.steps.characterReferences.form.inputs.firstName.id]:
  draftApplication.characterReferences?.[0]?.firstName ?? "",
```

Comment (SoLetsDev) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/wizard.ts:95]: 100%

File: src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/application.ts [hunk 3]
```diff
@@ -62,6 +62,25 @@ export const useApplicationStore = defineStore("application", {
...
       this.draftApplication.transcripts = Object.values(wizardStore.wizardData[wizardStore.wizardConfig.steps.education.form.inputs.educationList.id]);
+
+      // Check all required fields are present for valid character reference.
+      if (
+        !wizardStore.wizardData[wizardStore.wizardConfig.steps.characterReferences.form.inputs.firstName.id] ||
+        !wizardStore.wizardData[wizardStore.wizardConfig.steps.characterReferences.form.inputs.lastName.id] ||
+        !wizardStore.wizardData[wizardStore.wizardConfig.steps.characterReferences.form.inputs.email.id]
+      ) {
+        // Backend will not accept an invalid character reference. Save empty array of character references
+        this.draftApplication.characterReferences = [];
```
[the remark this answers is not in the document]
Comment (SoLetsDev) [reply src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/store/application.ts:73]: changed the flow. So we do not do anything with character references if there's missing required fields. 

If there's an existing character reference that information will not be wiped out.

**Review by pcatkins — APPROVED** [review]
Looks good — nice work!

**Pull request merged**